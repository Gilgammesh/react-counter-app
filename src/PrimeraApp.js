import React from "react";
import PropTypes from "prop-types";
//import React, {Fragment} from "react";

const PrimeraApp = ({ saludo }) => {
  //const saludo = "Hola Mundo";

  if (!saludo) {
    throw new Error("El saludo es necesario");
  }

  return (
    <>
      {/* <pre>{JSON.stringify(saludo, null, 3)}</pre> */}
      <h1>{saludo}</h1>
      <p>Mi primera aplicaci√≥n</p>
    </>
  );
};

PrimeraApp.propTypes = {
  saludo: PropTypes.string,
};

export default PrimeraApp;
