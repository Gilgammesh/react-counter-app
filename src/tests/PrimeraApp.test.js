import React from "react";
import { shallow } from "enzyme";
import "@testing-library/jest-dom";
import PrimeraApp from "../PrimeraApp";

describe("Pruebas en el componente PrimeraApp.js", () => {
  /************************************************************************************/
  /* test("se debe mostrar el saludo 'Hola mundo'", () => {
    const saludo = "Hola mundo";

    const { getByText } = render(<PrimeraApp saludo={saludo} />);

    expect(getByText(saludo)).toBeInTheDocument();
  }); */
  /************************************************************************************/
  test("se debe mostrar <PrimeraApp /> correctamente", () => {
    const saludo = "Hola mundo";

    const wrapper = shallow(<PrimeraApp saludo={saludo} />);

    expect(wrapper).toMatchSnapshot();
  });
  /************************************************************************************/
  test("se debe mostrar el subtitulo enviado en props", () => {
    const saludo = "Hola mundo";
    const subtitulo = "Este es el subtitulo de test";

    const wrapper = shallow(
      <PrimeraApp saludo={saludo} subtitulo={subtitulo} />
    );

    const textParrafo = wrapper.find("p").text();

    expect(textParrafo).toBe(subtitulo);
  });
});
