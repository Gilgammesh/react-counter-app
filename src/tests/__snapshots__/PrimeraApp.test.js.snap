// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente PrimeraApp.js se debe mostrar <PrimeraApp /> correctamente 1`] = `
<Fragment>
  <h1>
    Hola mundo
  </h1>
  <p>
    Soy un subtitulo
  </p>
</Fragment>
`;

exports[`Pruebas en el componente PrimeraApp.js se debe mostrar el subtitulo enviado en props 1`] = `
<Fragment>
  <h1>
    Hola mundo
  </h1>
  <p>
    Soy un subtitulo
  </p>
</Fragment>
`;
